### PR TITLE
fix issue #3245

### LINKED ISSUE #3245: Crash reported by Frosterx
## Issue explanation (write below this line)

Crash happened when trying to remove hidden files from google drive.

## Exception
* App Name: Amaze 文件管理器
* Package: com.amaze.filemanager
* Version: 3.6.7
* User Action: UI Error
* Request: Application crash
* OS: Linux Android 11 - 30
* Device: umi
* Model: Mi 10
* Product: umi
<details><summary><b>Crash log </b></summary><p>

`
java.lang.RuntimeException: The methods of CloudRail SI must not be executed on the UI Thread! They involve Thread blocking and network requests.
 at com.cloudrail.si.services.GoogleDrive.getMetadata(GoogleDrive.java:1529)
 at com.amaze.filemanager.filesystem.HybridFile.isDirectory(HybridFile.java:578)
 at com.amaze.filemanager.adapters.HiddenAdapter.lambda$onBindViewHolder$0(HiddenAdapter.java:112)
 at com.amaze.filemanager.adapters.HiddenAdapter.lambda$onBindViewHolder$0$HiddenAdapter(Unknown Source:0)
 at com.amaze.filemanager.adapters.-$$Lambda$HiddenAdapter$w3dqeqPOt1UdU

Reporter: Frosterx

### PR DESCRIPTION
**Pull request by happys2333**
## Description
Fixes https://github.com/TeamAmaze/AmazeFileManager/issues/3245

#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [x] Done  
- Device: OPPO A33m
- OS:Android 5.1.1 - 22

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

#### Array out of index bug fix
While I fixing this bug I found another bug in removing the hidden files. If you remove more than one file at one time, you will find the hiddenFiles array's index will get wrong. I found that the index was not updated at the time of deletion. So I added in the source code that every delete must be reindexed. If you do not add this, you will end up with an array out of bounds or removing the wrong file when you remove hidden files multiple times.

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 1]
```diff
@@ -572,11 +572,27 @@ public Boolean execute(SFTPClient client) {
...
+          public void run() {
+            isDir[0] = dataUtils
+                            .getAccount(OpenMode.GDRIVE)
+                            .getMetadata(CloudUtil.stripPath(OpenMode.GDRIVE, path))
+                            .getFolder();
+          }
+        });
+        googleThread.start();
+        try{
+          googleThread.join();
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:587]: No joining threads in main thread

**Review by EmmanuelMess — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java (hunk 1 above)
Comment (happys2333) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:587]: Any suggestions? I can't find a way to change it.

Comment (happys2333) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java]: The new code has update.

**Comment by VishalNehra** [comment]
please fix #3262 with this as well

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 2]
```diff
@@ -466,7 +466,7 @@ public String getParentName() {
 
   /**
    * Whether this object refers to a directory or file, handles all types of files
-   *
+   * Warning: Can't be used directly in UI/main thread
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:469]: This should be a `@WorkerThread` annotation.

**Review by EmmanuelMess — CHANGES_REQUESTED** [review]

**Comment by VishnuSanal** [comment]
@happys2333 this PR has no activity for too long. is this still salvageable? closing this for now; feel free to reopen since the issues this PR is trying to fix are still valid.

**Pull request closed without merging**